### PR TITLE
feat: [CI-6497]: delegateInfoList as initStepV2DelegateTaskInfo

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

Delegate selection logs will be placed in initStepV2DelegateTaskInfo for CI instead of delegateInfoList

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by shaurya-harness** [comment]
retrigger jest

**Comment by shaurya-harness** [comment]
retrigger coverage

**Review by d1wilko — APPROVED** [review]
LGTM

**Comment by shaurya-harness** [comment]
retrigger cypress-cv

File: src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx [hunk 1]
```diff
@@ -147,6 +147,32 @@ export function StepDetails(props: StepDetailsProps): React.ReactElement {
...
+                        vars={{ taskName: step.stepDetails?.initStepV2DelegateTaskInfo.taskName }}
+                        useRichText
+                      />
+                    </Text>{' '}
+                    (
+                    <Text
+                      font={{ size: 'small' }}
+                      onClick={() =>
+                        openDelegateSelectionLogsModal({
+                          taskId: step.stepDetails?.initStepV2DelegateTaskInfo.taskID as unknown as string,
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx:164]: `as unknown as string`
Can we just say `as string`?

Comment (shaurya-harness) [reply src/modules/70-pipeline/components/execution/StepDetails/common/StepDetails/StepDetails.tsx:164]: @ishantmehtaharness 
conflicts with 
```
stepDetails?: {
    [key: string]: {
      [key: string]: { [key: string]: any }
    }
```
coming from `src/services/pipeline-ng/index.tsx`

**Review by ishantmehtaharness — APPROVED** [review]

**Pull request merged**